### PR TITLE
[receiver/purefa] Adds the new scrapper type pods

### PR DESCRIPTION
**Pull request by dgoscn**
**Description:**
Adds the new scraper type to FlashArray receiver

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14886

**Testing:**
Only tests from the respective golang files

**Documentation:**
Still in progress

### CODE REVIEW

File: receiver/purefareceiver/receiver.go [hunk 1]
```diff
@@ -61,13 +61,20 @@ func (r *purefaReceiver) Start(ctx context.Context, compHost component.Host) err
...
 
-	directoriesScraper := internal.NewScraper(ctx, internal.ScraperTypeDirectories, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Host)
+	directoriesScraper := internal.NewScraper(ctx, internal.ScraperTypeDirectories, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Directories)
 	if scCfgs, err := directoriesScraper.ToPrometheusReceiverConfig(compHost, fact); err == nil {
 		scrapeCfgs = append(scrapeCfgs, scCfgs...)
 	} else {
 		return err
 	}
 
+	podsScraper := internal.NewScraper(ctx, internal.ScraperTypePods, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Pods)
```
Comment (jpkrohling) [inline receiver/purefareceiver/receiver.go:71]: r.cfg.Pods instead of Hosts

File: receiver/purefareceiver/receiver.go [hunk 2]
```diff
@@ -61,13 +61,20 @@ func (r *purefaReceiver) Start(ctx context.Context, compHost component.Host) err
 		return err
 	}
 
-	directoriesScraper := internal.NewScraper(ctx, internal.ScraperTypeDirectories, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Host)
+	directoriesScraper := internal.NewScraper(ctx, internal.ScraperTypeDirectories, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Directories)
```
Comment (jpkrohling) [inline receiver/purefareceiver/receiver.go:64]: Good catch!

**Review by jpkrohling — COMMENTED** [review]
Can you check if changed are needed for the readme and test data?

File: receiver/purefareceiver/receiver.go (hunk 1 above)
Comment (dgoscn) [reply receiver/purefareceiver/receiver.go:71]: @jpkrohling totally agree

**Comment by dgoscn** [comment]
> Can you check if changed are needed for the readme and test data?

Of course

File: receiver/purefareceiver/config.go [hunk 3]
```diff
@@ -42,16 +42,20 @@ type Config struct {
...
 	ReloadIntervals *ReloadIntervals `mapstructure:"reload_intervals"`
 }
 
 type ReloadIntervals struct {
-	Array       time.Duration `mapstructure:"array"`
-	Host        time.Duration `mapstructure:"host"`
-	Directories time.Duration `mapstructure:"directories"`
+	Array      time.Duration `mapstructure:"array"`
+	Host       time.Duration `mapstructure:"host"`
+	Directorie time.Duration `mapstructure:"directorie"`
```
Comment (jpkrohling) [inline receiver/purefareceiver/config.go:57]:
```suggestion
	Directories time.Duration `mapstructure:"directories"`
```

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**